### DISCUSSION TITLE
Support for both USB peripherals on STM32

### DISCUSSION THREAD

**Discussion by rechrtb** [hathach/tinyusb#2019] [Q&A]
Some STM32 have two USB peripherals, one FS OTG and another HS OTG. In my case, I'm going to be using them both in FS device mode only. I was wondering if tinyUSB supports multiple 'instances' of itself, one for each peripheral, operating simultaneously.

The reason I ask is that the [documentation](https://docs.tinyusb.org/en/latest/reference/getting_started.html#port-selection) seem to imply that only one can be supported at a time. To quote:

>If a board has several ports, one port is chosen by default in the individual board.mk file. Use option PORT=x To choose another port. For example to select the HS port of a STM32F746Disco board, use:
>
>$ make BOARD=stm32f746disco PORT=1 all

**Comment by hathach** [comment]
tinyusb does not support multiple device ports, only host ports. It is actually dropped quite some time ago for simpler implementation. But it is possible in term of API/callback if you want to get your hand dirty. Though you would have to do a significant works to get it running.

**Reply by rechrtb** [reply]
I see, thanks for the clarification!
